pkg/storage/s3.go

```diff
@@ -75,12 +75,12 @@ type S3Uploader struct {
 
 // UploadPart update partial data to s3, we should call CreateMultipartUpload to start it,
 // and call CompleteMultipartUpload to finish it.
-func (u *S3Uploader) UploadPart(ctx context.Context, data []byte, partNum int) error {
+func (u *S3Uploader) UploadPart(ctx context.Context, data []byte) error {
 	partInput := &s3.UploadPartInput{
 		Body:          bytes.NewReader(data),
 		Bucket:        u.createOutput.Bucket,
 		Key:           u.createOutput.Key,
-		PartNumber:    aws.Int64(int64(partNum)),
+		PartNumber:    aws.Int64(int64(len(u.completeParts))),
```

Comment:
```suggestion
		PartNumber:    aws.Int64(int64(len(u.completeParts) + 1)),
```

https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html

> When uploading a part, in addition to the upload ID, you must specify a part number. You can choose any part number **between 1 and 10,000**.

0 is not a valid part number

Reply:
fixed